---
title: "Phosphoproteomics pipeline using PlexedPiper"
author: "Michael Nestor"
date: "4/7/2021"
output: rmarkdown::github_document
---

# Global Proteomics analysis

The global proteomics pipeline uses the R package [PlexedPiper](https://github.com/vladpetyuk/PlexedPiper). It also requires a connection to the DMS to access data packages.

```{r, echo=F}
library(knitr)
knitr::opts_chunk$set(message=F, warning=F)
```

```{r}
library(PlexedPiper)
data_package_num <- 3626

if (!is_PNNL_DMS_connection_successful()) {
  stop("No connection to DMS.")
}
```

## 1. Read study design information

Study design information in PlexedPiper is encoded in three tables: fractions, samples, and references. These tables can be made using metadata and should be stored on the DMS before processing.

```{r}
study_design <- get_study_design_by_dataset_package(data_package_num)

fractions <- study_design$fractions
samples <- study_design$samples
references <- study_design$references
```

## 2 Processing MS-GF+ data

MS-GF+ data is processed in several steps. First, read MS-GF+ output from the DMS. (This step can take a while).

```{r}
msnid <- read_msms_data_from_DMS(data_package_num)
show(msnid)
```

### 2.1 AScore

```{r}
ascore <- get_AScore_results(data_package_num)
msnid <- best_PTM_location_by_ascore(msnid, ascore)
msnid <- apply_filter(msnid, "grepl(\"\\\\*\", peptide)")
```

### 2.1 Remap accessions

This function remaps UniProt protein accessions to gene symbol.

```{r}
msnid <- remap_accessions_refseq_to_gene(msnid,
                                          organism_name="Rattus norvegicus")
show(msnid)
```

### 2.2 FDR filter

We use the target-decoy search strategy method described in [(Elias 2010)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2922680/). Filtering is done first at peptide level, then at protein level, both with max FDR of 1%.

```{r}
msnid <- filter_msgf_data_peptide_level(msnid, 0.01)
show(msnid)
```

```{r}
msnid <- apply_filter(msnid, "!isDecoy")
show(msnid)
```


### 2.3 Parsimonious inference

To reduce number of protein identifications, we use a parsimonious inference algorithm described in [(Zhang et al. 2007)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2810678/).

```{r}
# global_crosstab <- ###
# global_proteins <- rownames(global_crosstab)
msnid <- infer_parsimonious_accessions(msnid,
                                       prior = global_proteins)
show(msnid)
```

### Map mod sites

```{r}
path_to_FASTA <- path_to_FASTA_used_by_DMS(data_package_num)
path_to_FASTA <- gsub("\\\\","/",path_to_FASTA)
path_to_FASTA_gene <- remap_accessions_uniprot_to_gene_fasta(path_to_FASTA)

library(Biostrings)
fst <- readAAStringSet(path_to_FASTA_gene)
msnid <- map_mod_sites(msnid, fst, 
                       accession_col = "accession", 
                       peptide_mod_col = "peptide", 
                       mod_char = "*",
                       site_delimiter = "lower")
msnid <- apply_filter(msnid, "!is.na(PepLocFirst)")
head(psms(msnid))
```

## 3 Process MASIC data

Output from the MASIC software is read from DMS, then filtered by inteference score.

```{r}
masic_data <- read_masic_data_from_DMS(data_package_num,
                                       interference_score = TRUE)
nrow(masic_data)
```

```{r}
masic_data <- filter_masic_data(masic_data,
                                interference_score_threshold = 0.5,
                                s2n_threshold = 0)
nrow(masic_data)
```

## 4 Create crosstab

The quantitative crosstab combines MS/MS identifications with reporter ion intensities. Abundances are taken relative to the reference channel and then log-transformed.

```{r}
aggregation_level <- c("SiteID")
crosstab <- create_crosstab(msnid, masic_data,
                            aggregation_level,
                            fractions, samples, references)

head(crosstab)
```